### PR TITLE
Harden image

### PR DESCRIPTION
**Pull request by dandersonsw**
## Changes proposed in this pull request:

- Updates dockerfile to use our hardened base image
- Updates pipeline to use the hardened testing image

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations

Updating code to use hardened image

### CODE REVIEW

File: ci/pipeline.yml
Comment (markdboyd) [inline ci/pipeline.yml:1]: Can we split the PR to do this file change and the change to the Dockerfile separately? Once we merge this change, the pipeline is broken until the hardened image is available. Whereas the change to the Dockerfile can be done anytime

Comment (dandersonsw) [reply ci/pipeline.yml:1]: I think the pipeline is going to break either way, because once the change to the Dockerfile is made, the pipeline is going to try to build the docker image, but it won't be able to because there isn't anything set to populate the base_image arg in the dockerfile.

Comment (markdboyd) [reply ci/pipeline.yml:1]: OK. I might hold off on approving/merging until I get at least one more fix in for the pipeline then, if that's okay

Comment (dandersonsw) [reply ci/pipeline.yml:1]: Yep, no problem. It might be good to wait and roll out a change like this next week anyways

**Review by markdboyd — APPROVED** [review]

**Pull request merged**
